draft-ietf-masque-connect-ip.md

```diff
@@ -1273,6 +1279,58 @@ ADDRESS_ASSIGN capsule is parsed. This allows modifications to address
 assignement to operate atomically. Similarly, extensions that modify routing
 SHOULD behave similarly with regards to the ROUTE_ADVERTISEMENT capsule.
 
+# Performance Considerations
+
+Bursty traffic can often lead to temporally-correlated packet losses; in turn,
+this can lead to suboptimal responses from congestion controllers in protocols
+running inside the tunnel. To avoid this, endpoints SHOULD strive to avoid
+increasing burstiness of IP traffic; they SHOULD NOT queue packets in order to
+increase batching beyond the minimal amount required to take advantage of
+hardware offloads.
+
+When the protocol running inside the tunnel uses congestion control (e.g.,
+{{TCP}} or {{QUIC}}), the proxied traffic will incur at least two nested
+congestion controllers. The outer HTTP connection MUST NOT disable congestion
+control unless it has an out-of-band way of knowing with absolute certainty
+that all of the inner packets belong to congestion-controlled connections.
```

Comment:
I have to say that I share Bob's view that this requirement is unnecessarily strict for IP traffic. The general assumption is that IP traffic will be congestion controlled. Some will not be, but that doesn't change things from a tunnel perspective. I would in fact actually propose that we change this sentence to:

The outer HTTP connection MAY disable congestion
 control for QUIC packets that primarily contain tunneled IP packets sent using datagrams, this as IP traffic can be assumed to be congestion controlled end-to-end.

Reply:
I flipped the sentence to replace MUST NOT with MAY